docs/src/routes/docs/[...3]modules/transaction-preview.md

```diff
@@ -73,6 +73,102 @@ const onboard = Onboard({
+
+```typescript
+import transactionPreviewModule from '@web3-onboard/transaction-preview'
+
+const {init, previewTransaction} = transactionPreviewModule({
+  // Optional: Require balance change approval prior to sending transaction to wallet
+  // Defaults to true
+  // requireTransactionApproval?: false
+
+  //  i18n?: i18nOptions - Internationalization options
+})
+await init({
+/**
+ * Blocknative API key (https://explorer.blocknative.com/account)
+ */
+apiKey: string
+/**
+ * Your Blocknative SDK instance
+ * */
+sdk: SDK
+/**
+ * Optional dom query string to mount UI to
+ * */
+containerElement: string})
+
+// Transaction code here using Ether.js or Web3.js or construct your own transactions
+const simulate = async provider => {
+  const ethersProvider = new ethers.providers.Web3Provider(provider, 'any')
+
+  const signer = ethersProvider.getSigner()
+  const addressFrom = '0xcxxxxxx11111999991111'
+
+  // Uniswap V2
+  const CONTRACT_ADDRESS = '0x7a250d5630b4cf539739df2c5dacb4c659f2488d'
+  const erc20_interface = [
+    'function approve(address _spender, uint256 _value) public returns (bool success)',
+    'function transferFrom(address sender, address recipient, uint256 amount) external returns (bool)',
+    'function balanceOf(address owner) view returns (uint256)'
+  ]
+
+  const uniswapV2router_interface = [
+    'function swapExactTokensForETH(uint amountIn, uint amountOutMin, address[] calldata path, address to, uint deadline) external returns (uint[] memory amounts)'
+  ]
+
+  const weth = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'
+  const oneInch = '0x111111111117dc0aa78b770fa6a738034120c302'
+  let swapTxData
+  let approveTxData
+  const swapContract = new ethers.Contract(
+    CONTRACT_ADDRESS,
+    uniswapV2router_interface
+  )
+  const erc20_contract = new ethers.Contract(oneInch, erc20_interface)
+  const oneEther = ethers.BigNumber.from('9000000000000000000')
+  approveTxData = await erc20_contract.populateTransaction.approve(
+    CONTRACT_ADDRESS,
+    oneEther
+  )
+
+  const amountOutMin = 0
+  const amountOutMinHex = ethers.BigNumber.from(amountOutMin.toString())._hex
+
+  const path = [oneInch, weth]
+  const deadline = Math.floor(Date.now() / 1000) + 60 * 1 // 1 minutes from the current Unix time
+
+  const inputAmountHex = oneEther.toHexString()
+
+  swapTxData = await swapContract.populateTransaction.swapExactTokensForETH(
+    inputAmountHex,
+    amountOutMinHex,
+    path,
+    addressFrom,
+    deadline
+  )
+  const uniswapV2Router = '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D'
+
+  const popApproveTransaction = await signer.populateTransaction(approveTxData)
+  const popTransaction = await signer.populateTransaction(swapTxData)
+  const transactions = [
+    { ...popApprovedTransaction, value: 0 },
+    {
+      ...popTransaction,
+      from: addressFrom,
+      to: uniswapV2Router,
+      value: 0
+    }
+  ]
+  await previewTransaction(transactions)
+}
+
+simulate(ethereumProvider)
```

Comment:
Does the `previewTransaction` function return the simulated payload? If so, i reckon the example should show a response being returned ie: `const simulated = await simulate(ethereumProvider)`

Reply:
It does not return the payload it will render the TP modal to the containerEl passed into the init - do you think it would be valuable to return the whole payload as well?
I had a similar thought but was kind of on the fence around that

Reply:
Added `MultiSimOutput` response to `previewTransaction` function and updated this example....thanks guys!